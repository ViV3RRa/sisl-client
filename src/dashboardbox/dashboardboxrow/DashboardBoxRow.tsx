import React, { FC } from 'react';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import IconButton from '@material-ui/core/IconButton';
import EditIcon from '@material-ui/icons/Edit';
import DeleteIcon from '@material-ui/icons/Delete';
import { SislIcon } from '../../icons/sislIcon/SislIcon';
import { AccountValue } from '../../customtypes';

interface DashboardBoxRowProps {
  accountValue: AccountValue;
}

export const DashboardBoxRow: FC<DashboardBoxRowProps> = (
  props: DashboardBoxRowProps
) => {
  let actions: any;
  if (false) {
    actions = (
      <TableCell align="right">
        <IconButton aria-label="Rediger">
          <SislIcon muiIcon={EditIcon} />
        </IconButton>
      </TableCell>
    );
  } else if (false) {
    actions = (
      <TableCell align="right">
        <IconButton aria-label="Slet">
          <SislIcon muiIcon={DeleteIcon} />
        </IconButton>
      </TableCell>
    );
  } else {
    actions = <TableCell align="right"></TableCell>;
  }

  return (
    <React.Fragment>
      <TableRow>
        <TableCell>{props.accountValue.accountValue}</TableCell>
        {actions}
      </TableRow>
    </React.Fragment>
  );
};
