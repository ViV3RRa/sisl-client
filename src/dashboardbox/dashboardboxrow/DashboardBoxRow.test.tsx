import React from 'react';
import { shallow } from 'enzyme';
import { DashboardBoxRow } from './DashboardBoxRow';

it('renders without crashing (enzyme)', () => {
  const result = shallow(
    <DashboardBoxRow obj={{ type: '', title: '', values: [], action: '' }} />,
    {
      context: {}
    }
  );
  expect(result.debug()).toMatchSnapshot();
});
