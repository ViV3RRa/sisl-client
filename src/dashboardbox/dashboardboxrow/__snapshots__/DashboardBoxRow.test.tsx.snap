// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing (enzyme) 1`] = `
"<Fragment>
  <WithStyles(ForwardRef(TableRow))>
    <WithStyles(ForwardRef(TableCell)) />
    <WithStyles(ForwardRef(TableCell)) align=\\"right\\" />
  </WithStyles(ForwardRef(TableRow))>
</Fragment>"
`;
